src/wp-includes/class-wp-user-query.php

```diff
@@ -1116,6 +1116,18 @@ protected function parse_order( $order ) {
 	public function __get( $name ) {
 		if ( in_array( $name, $this->compat_fields, true ) ) {
 			return $this->$name;
+		} else {
+			// Alert and inform developers about the incorrect use of dynamic property
+			_doing_it_wrong(
```

Comment:
Instead of `_doing_it_wrong()`, I think `trigger_error()` with a `E_USER_DEPRECATION` is a better choice. Why?

>The PHP 8.2 Dynamic Property change triggers a `E_USER_DEPRECATED`, whereas `_doing_it_wrong()` triggers a E_USER_NOTICE.

With `trigger_error()`, `__()` translation is not required.